fly/eventstream/render.go

```diff
@@ -75,14 +81,17 @@ func Render(dst io.Writer, src eventstream.EventStream) int {
 					exitStatus = 3
 				}
 			default:
-				fmt.Fprintf(dst, "unknown status: %s", e.Status)
+				eventLog = NewEventLogFromStatus("unknown status: %s",  e.Status, e.Time)
+				fmt.Fprint(dst, AdditionalFormatting(eventLog, options))
 				return 255
 			}
 
 			printColorFunc := printColor.SprintFunc()
-			fmt.Fprintf(dst, "%s\n", printColorFunc(e.Status))
+			eventLog = NewEventLog("%s\n", printColorFunc(e.Status), e.Time)
+			fmt.Fprint(dst, AdditionalFormatting(eventLog, options))
 
 			return exitStatus
 		}
+		fmt.Fprint(dst, AdditionalFormatting(eventLog, options))
```

Comment:
I think this would be less code if you wrapped `dst` with a single writer that kept track of the timestamp and conditionally added it at the start of each line. Sort of like the `PrefixedWriter`, but stateful and more domain-specific. Then all the `fmt.Printf`s don't have to change and you don't need the intermediary `EventLog` type.

Something like:

```go
dst := NewTimestampedWriter(dst, options.ShowTimestamps)

for {
  // ...
  switch e := ev.(type) {
  case event.Log:
    dst.SetTimestamp(e.Time)
    fmt.Fprint(dst, "%s", e.Payload)
  // ...
  }
}
```

I think the current approach works fine, just a suggestion. 🙂 I'm being especially careful with `fly` PRs because it's long overdue for some major refactoring in other areas. Hope that's OK!

Reply:
Updated it to be the way you visioned!